pkg/cmd/repo/setdefault/setdefault.go

```diff
@@ -121,12 +121,13 @@ func setDefaultRun(opts *SetDefaultOptions) error {
 	if opts.ViewMode {
 		if currentDefaultRepo != nil {
 			fmt.Fprintln(opts.IO.Out, displayRemoteRepoName(currentDefaultRepo))
+			return nil
 		} else if opts.IO.IsStdoutTTY() {
-			fmt.Fprintln(opts.IO.Out, "no default repository has been set; use `gh repo set-default` to select one")
+			fmt.Fprintln(opts.IO.ErrOut, "no default repository has been set; use `gh repo set-default` to select one")
+			return cmdutil.SilentError
 		}
-		return nil
+		return cmdutil.SilentError
```

Comment:
I think we want to continue to return `nil`, here. In fact, I think we can remove the `else` entirely:

The command isn't failing, so we aren't returning an error. That would introduce a breaking change. Instead, we are routing the messaging to `stderr` instead of `stdout` so that we log to the user _why_ this isn't returning anything without breaking any piping or automation experiences.

Reply:
> _The command isn't failing, so we aren't returning an error._

To dig into this deeper, `stderr` isn't solely used for telling users about errors; CLIs trying to avoid breaking Unix command output / input piping use it for a whole variety of reasons.

Additionally, `gh` has a predictable approach to determining the base repo in various situations while giving users the ability to override this logic using `gh repo set-default`.  It isn't an error if the user has not explicitly overrode the base repo.